pkg/daemon/update.go

```diff
@@ -382,6 +389,40 @@ func verifyUserFields(pwdUser ignv2_2types.PasswdUser) error {
 	return nil
 }
 
+func generateKargsCommand(oldConfig, newConfig *mcfgv1.MachineConfig) []string {
+	oldKargs := make(map[string]bool)
+	for _, arg := range oldConfig.Spec.KernelArguments {
+		oldKargs[arg] = true
+	}
+	newKargs := make(map[string]bool)
+	for _, arg := range newConfig.Spec.KernelArguments {
+		newKargs[arg] = true
+	}
+	cmdArgs := []string{}
+	for arg := range oldKargs {
+		if !newKargs[arg] {
+			cmdArgs = append(cmdArgs, "--delete=" + arg)
+		}
+	}
+	for arg := range newKargs {
```

Comment:
just a note that ranging over a map in golang isn't consistent wrt ordering to that `cmdArgs` can have a different kargs order than what we have in the newConfig.Spec.KernelArguments. If it's important we should account for that

Reply:
Hmm...I *think* this could only matter if one had multiply specified/conflicting arguments, like `mem=4096 mem=8192` or something...I think like most argument parsers the kernel is "last one wins" so indeed in that scenario it could be unpredictable.

I'd probably say "just don't do that"?  But we can also try to handle it.

Reply:
`console=` is also another example where ordering matters right? handling it would be easy also, we're not expecting huge kargrs lists. We can use a slice with more logic instead of a map[string]bool to maintain consisten order (again, the kargs slice isn't huge so we don't really care about O(1) when checking if an arg is there or not, right?)

Reply:
I reworked this to iterate explicitly over the new args.  Though probably the testing here for order is better done in an e2e fashion because there's a lot of code in the rpm-ostree/ostree stack here processing this too.

Reply:
There are libostree-level issues to fix here https://github.com/ostreedev/ostree/issues/1859
Eventually we'll have to get this right through the whole stack, but I think what we're doing here in the MCD is OK for now.